### PR TITLE
Msue 40 verification end points

### PR DESCRIPTION
**Pull request by jyothish6190**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,8 +153,61 @@ try:
 except sift.client.ApiException:
     # request failed
     pass
-```
 
+# The send call  triggers the generation of a OTP code that is stored by Sift and emails the code to the user.
+
+send_properties = {
+            "$user_id": "billy_jones_301",
+            "$send_to":	"billy_jones_301@gmail.com",
```
Comment (mlunney) [inline README.md:161]: you can delete the extra spaces between the key and value here.

File: sift/client.py [hunk 2]
```diff
@@ -18,8 +20,17 @@
...
 API_URL = 'https://api.siftscience.com'
 API3_URL = 'https://api3.siftscience.com'
+API_URL_VERIFICATION = 'https://api.sift.com/v1.1/verification/'
+
 DECISION_SOURCES = ['MANUAL_REVIEW', 'AUTOMATED_RULE', 'CHARGEBACK']
 
+VERIFIED_EVENTS = [
+    "$add_item_to_cart", "$add_promotion", "$content_status",
+    "$create_account", "$create_content", "$create_order", "$flag_content",
+    "$login", "$order_status", "$remove_item_from_cart", "$transaction", "$transaction",
```
Comment (mlunney) [inline sift/client.py:30]: You have `$transaction` twice.

File: sift/client.py [hunk 3]
```diff
@@ -708,7 +725,128 @@ def apply_content_decision(self, user_id, content_id, properties, timeout=None):
...
+
+        user_id = properties.get('$user_id')
+        _assert_non_empty_unicode(user_id, 'user_id', error_cls=ValueError)
+
+        otp_code = properties.get('$code')
+        if otp_code is None:
+            raise ValueError("Code is required")
+
+        if otp_code and type(otp_code) != int:
+            raise ValueError("Code should be  a number")
```
Comment (mlunney) [inline sift/client.py:801]: Extra space between "be" and "a".
I think the code should be a String. It's possible to have leading zeroes in the OTP which is a problem.

File: sift/client.py [hunk 4]
```diff
@@ -708,7 +725,128 @@ def apply_content_decision(self, user_id, content_id, properties, timeout=None):
...
+
+        verified_event = properties.get('$verified_event')
+        if (verified_event and verified_event not in VERIFIED_EVENTS):
+            raise ValueError(" 'verified_event' must be one of [{0}]".format(
```
Comment (mlunney) [inline sift/client.py:805]: Do we need this check? I think the server returns an error if the event type is wrong.

You have an extra space at start of the ValueError message.

File: sift/client.py [hunk 5]
```diff
@@ -708,7 +725,128 @@ def apply_content_decision(self, user_id, content_id, properties, timeout=None):
...
+                ", ".join(VERIFIED_EVENTS)))
+
+        verified_entity_id = properties.get('$verified_entity_id')
+        _assert_non_empty_unicode(
+            verified_entity_id, 'verified_entity_id', error_cls=ValueError)
```
Comment (mlunney) [inline sift/client.py:810]: $verified_entity_id is optional.

File: sift/client.py [hunk 6]
```diff
@@ -723,6 +861,101 @@ def apply_content_decision(self, user_id, content_id, properties, timeout=None):
...
+        _assert_non_empty_unicode(user_id, 'user_id', error_cls=ValueError)
+
+        verified_event = properties.get('$verified_event')
+        if (verified_event and verified_event not in VERIFIED_EVENTS):
+            raise ValueError(" 'verified_event' must be one of [{0}]".format(
+                ", ".join(VERIFIED_EVENTS)))
+
+        verified_entity_id = properties.get('$verified_entity_id')
+        _assert_non_empty_unicode(
+            verified_entity_id, 'verified_entity_id', error_cls=ValueError)
```
Comment (mlunney) [inline sift/client.py:957]: verified_entity_id is optional.

File: tests/test_client.py [hunk 7]
```diff
@@ -1196,6 +1237,119 @@ def test_provided_session(self):
...
+                auth=mock.ANY, data=data, headers=mock.ANY, timeout=mock.ANY)
+            self.assertIsInstance(response, sift.client.Response)
+            assert(response.is_ok())
+            assert(response.body['error_message'] == 'OK')
+
+    def test_send_ok(self):
+        mock_response = mock.Mock()
+        send_request = {
+            "$user_id": "billy_jones_301",
+            "$send_to":	"billy_jones_301@gmail.com",
```
Comment (mlunney) [inline tests/test_client.py:1277]: Remove extra space between key/value.

**Comment by viaskal-sift** [comment]
@mlunney do we still need this one? Seems there is another PR already merged which is probably contains changes related (at least based on Jira linkage)

**Pull request closed without merging**